tests/test-desktop-files.lua

```diff
@@ -0,0 +1,34 @@
+local runner = require("_runner")
+local menubar = require("menubar")
+local gears_debug = require("gears.debug")
+
+local assets_dir = os.getenv("TEST_ASSETS") .. "/"
+menubar.menu_gen.all_menu_dirs = { assets_dir }
+
+local menu_gen_list = nil
+
+runner.run_steps({
+    function()
+        menubar.menu_gen.generate(function(menu_entries)
+            menu_gen_list = menu_entries
+        end)
+        return true
+    end,
+    function(count)
+        if menu_gen_list == nil then
+            -- Continue waiting
+            if count >= 5 then
+                return false
+            end
+            return
+        else
+            -- Begin the actual test
+            local active_entry = menu_gen_list[1] -- Lua array counting starts at 1, right?
+            assert(active_entry.icon == "/usr/share/icons/hicolor/128x128/apps/firefox.png")
```

Comment:
It's peppered across several commits - I'll re-squash commits around so things make more sense, but the file itself is [here](https://github.com/Crazylemon64/awesome/blob/be7d02b56ea4d18387ca55d299d2e0a0b02ae116/tests/assets/example_1.desktop). Last I looked over icon path generation code, it didn't actually require any icons to be installed on the system, but I'll give another look-over to make certain of that.